embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/internal/logs/LogOrchestrator.kt

```diff
@@ -0,0 +1,78 @@
+package io.embrace.android.embracesdk.internal.logs
+
+import androidx.annotation.VisibleForTesting
+import io.embrace.android.embracesdk.comms.delivery.DeliveryService
+import io.embrace.android.embracesdk.internal.clock.Clock
+import io.embrace.android.embracesdk.worker.ScheduledWorker
+import java.util.concurrent.ScheduledFuture
+import java.util.concurrent.TimeUnit
+
+internal class LogOrchestrator(
+    private val logOrchestratorScheduledWorker: ScheduledWorker,
+    private val clock: Clock,
+    private val sink: LogSinkImpl,
+    private val deliveryService: DeliveryService
+) {
+    private var lastLogTime: Long = 0
+    private var inactivityFuture: ScheduledFuture<*>? = null
+    private var batchTimeFuture: ScheduledFuture<*>? = null
+
+    init {
+        sink.callOnLogsStored(::onLogsAdded)
+    }
+    private fun onLogsAdded() {
+        lastLogTime = clock.now()
+        if (sink.completedLogs().size >= MAX_LOGS_PER_BATCH) {
+            sendLogs()
+        } else {
+            scheduleInactivityCheck()
+            if (batchTimeFuture == null) {
+                scheduleBatchTimeCheck()
+            }
+        }
+    }
+
+    @VisibleForTesting
```

Comment:
We probably shouldn't expose methods for testing purposes. Instead of observing the internals of this method and what it does, you can provide a fake `LogSink` so you can observe the data that gets written there via this method.

You can have a look at `SpanServiceImplTest` to see how that can be done.